test/livebook/live_markdown/import_test.exs

```diff
@@ -766,4 +766,82 @@ defmodule Livebook.LiveMarkdown.ImportTest do
              } = notebook
     end
   end
+
+  test "import notebook with invalid parent section sets parent_id of section to nil" do
+    markdown = """
+    # My Notebook
+
+    <!-- livebook:{"branch_parent_index":4} -->
+
+    ## Section 3
+
+    ```elixir
+    Process.info()
+    ```
+
+    ## Section 4
+
+    ```elixir
+    Process.info()
+    ```
+    """
+
+    {notebook,
+     [
+       "[warning] Section [Section 3] has a parent section which is either after the section it self or its parent is a branching section"
+     ]} = Import.notebook_from_markdown(markdown)
+
+    assert %Notebook{
+             name: "My Notebook",
+             sections: [
+               %Notebook.Section{
+                 name: "Section 3",
+                 parent_id: nil,
+                 cells: [
+                   %Cell.Elixir{
+                     source: """
+                     Process.info()\
+                     """
+                   }
+                 ]
+               },
+               %Notebook.Section{
+                 name: "Section 4",
+                 parent_id: _,
+                 cells: [
+                   %Cell.Elixir{
+                     source: """
+                     Process.info()\
+                     """
+                   }
+                 ]
+               }
+             ]
+           } = notebook
+  end
+
+  test "import notebook with invalid parent section produces a warning message" do
+    markdown = """
+    # My Notebook
+
+    <!-- livebook:{"branch_parent_index":4} -->
+
+    ## Section 3
+
+    ```elixir
+    Process.info()
+    ```
+
+    ## Section 4
+
+    ```elixir
+    Process.info()
+    ```
+    """
+
+    assert {_notebook,
+            [
+              "[warning] Section [Section 3] has a parent section which is either after the section it self or its parent is a branching section"
+            ]} = Import.notebook_from_markdown(markdown)
+  end
```

Comment:
This test is already covered above, and I think it's fine to assert both in the same test. Then as we will have two warnings, we can have on test per each :)